### PR TITLE
WRN-848: Updated to use `forwardCustom` and add `type` when forwarding custom events

### PR DESCRIPTION
**Pull request by MikyungKim**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Some custom events don't have `type` property.
We've introduced `forwardCustom` for forwarding custom events.

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
Updated to use `forwardCustom` and add `type` where it needs.

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)
Some private components are not updated.

### Links
[//]: # (Related issues, references)
WRN-848

### Comments
Enact-DCO-1.0-Signed-off-by: Mikyung Kim (mikyung27.kim@lge.com)

### CODE REVIEW

**Comment by 0x64** [comment]
I found `ContextualPopupDecorator` also has `forward` calls of `onClose` for key events. It could be better to change them also. Thank you!

File: Panels/Header.js [hunk 1]
```diff
@@ -390,6 +391,10 @@ const HeaderBase = kind({
 		}
 	},
 
+	handlers: {
+		onBack: forwardCustom('onBack')
```
Comment (0x64) [inline Panels/Header.js:395]: Let's add `onClose` handler also to be used for a close button.

Comment (MikyungKim) [reply Panels/Header.js:395]: 🆗

File: .travis.yml [hunk 2]
```diff
@@ -7,7 +7,7 @@ install:
     - npm config set prefer-offline true
     - npm install -g enactjs/cli#develop
     - npm install -g codecov
-    - git clone --branch=develop --depth 1 https://github.com/enactjs/enact ../enact
+    - git clone --branch=feature/WRN-848 --depth 1 https://github.com/enactjs/enact ../enact
```
Comment (MikyungKim) [inline .travis.yml:10]: We need to roll back this before merging.

**Review by 0x64 — APPROVED** [review]

File: DayPicker/DaySelectorDecorator.js [hunk 3]
```diff
@@ -275,7 +275,7 @@ const DaySelectorDecorator = hoc((config, Wrapped) => {
 			selected = generalizeSelected(selected, state);
 			const content = getSelectedDayString(selected, '', dayNameLength);
 
-			forwardSelect({selected, content}, this.props);
+			forwardSelect({type: 'onSelect', selected, content}, this.props);
```
[the remark this answers is not in the document]
Comment (MikyungKim) [reply DayPicker/DaySelectorDecorator.js:278]: I agree.

**Pull request merged**